src/execution_tree/compiler/compiler.cpp

```diff
@@ -443,11 +434,10 @@ namespace phylanx { namespace execution_tree { namespace compiler
                         patterns_, default_locality_));
                 }
 
-                if (id.id >= 0)
-                {
-                    name += annotation(id);
-                }
-                return (*cf)(std::move(args), name, name_);
+                primitive_name_parts name_parts{std::move(name), -1, id.id,
```

Comment:
Please be careful to take an empty `tagged_id` into account.

Reply:
Empty `tagged_id`s have their `id` and `col` set to `-1`. No problem here.